src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,121 @@
+import classnames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  onSubmit: (data: CommentData & { postId: number }) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onSubmit }) => {
+  const [formState, setFormState] = useState({
+    formData: {
+      name: '',
+      email: '',
+      body: '',
+    },
+    errors: {
+      name: false,
+      email: {
+        empty: false,
+        invalid: false,
+      },
+      body: false,
+    },
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = e.target;
+
+    setFormState(currState => ({
+      ...currState,
+      formData: {
+        ...currState.formData,
+        [name]: value,
+      },
+      errors: {
+        ...currState.errors,
+        [name]: name === 'email' ? { empty: false, invalid: false } : false,
+      },
+    }));
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const trimmedFormData = {
+      name: formState.formData.name.trim(),
+      email: formState.formData.email.trim(),
+      body: formState.formData.body.trim(),
+    };
+
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    const newErrors = {
+      name: !trimmedFormData.name,
+      email: {
+        empty: !trimmedFormData.email,
+        invalid:
+          !!trimmedFormData.email && !emailRegex.test(trimmedFormData.email),
+      },
+      body: !trimmedFormData.body,
+    };
+
+    setFormState(currState => ({
+      ...currState,
+      errors: newErrors,
+    }));
+
+    if (
+      newErrors.name ||
+      newErrors.email.empty ||
+      newErrors.email.invalid ||
+      newErrors.body
+    ) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      await onSubmit({ ...trimmedFormData, postId });
+
+      setFormState(currState => ({
+        ...currState,
+        formData: {
+          ...currState.formData,
+          body: '',
+        },
+      }));
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleClear = () => {
```

Comment:
Good job on handling form reset. This improves user experience as they can easily clear the form.